Mathlib/FieldTheory/Normal.lean

```diff
@@ -177,6 +177,22 @@ instance normal_iSup {ι : Type*} (t : ι → IntermediateField F K) [h : ∀ i,
   exact Polynomial.splits_comp_of_splits _ (inclusion hE).toRingHom this
 #align intermediate_field.normal_supr IntermediateField.normal_iSup
 
+/-- If a set of algebraic elements in a field extension `K/F` have minimal polynomials that
+  split in another extension `L/F`, then all minimal polynomials in the intermediate field
+  generated by the set also split in `L/F`. -/
+theorem splits_of_mem_adjoin {L} [Field L] [Algebra F L] {S : Set K}
+    (splits : ∀ x ∈ S, IsIntegral F x ∧ (minpoly F x).Splits (algebraMap F L)) {x : K}
+    (hx : x ∈ adjoin F S) : (minpoly F x).Splits (algebraMap F L) := by
+  let E : IntermediateField F L := ⨆ x : S, adjoin F ((minpoly F x.val).rootSet L)
+  have normal : Normal F E := normal_iSup (h := fun x ↦
+    Normal.of_isSplittingField (hFEp := adjoin_rootSet_isSplittingField (splits x x.2).2))
+  have : ∀ x ∈ S, (minpoly F x).Splits (algebraMap F E) := fun x hx ↦ splits_of_splits
+    (splits x hx).2 fun y hy ↦ (le_iSup _ ⟨x, hx⟩ : _ ≤ E) (subset_adjoin F _ <| by exact hy)
+  obtain ⟨φ⟩ := nonempty_algHom_adjoin_of_splits fun x hx ↦ ⟨(splits x hx).1, this x hx⟩
+  convert splits_comp_of_splits _ E.val.toRingHom (normal.splits <| φ ⟨x, hx⟩)
+  rw [minpoly.algHom_eq _ φ.injective, ← minpoly.algHom_eq _ (adjoin F S).val.injective]
+  rfl
```

Comment:
I haven't looked at this super closely, but it feels like this should all be easier. For instance, going from `(minpoly F x).Splits (algebraMap F E)` to `(minpoly F x).Splits (algebraMap F L)` feels like it should be a one-liner. Are there missing lemmas in the library perhaps?

Reply:
> going from `(minpoly F x).Splits (algebraMap F E)` to `(minpoly F x).Splits (algebraMap F L)` feels like it should be a one-liner.

I think it's [Polynomial.splits_comp_of_splits](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Polynomial/Splits.html#Polynomial.splits_comp_of_splits)

Reply:
As @acmepjz said, if the two `minpoly F x` were defeq then `convert splits_comp_of_splits` would close the goal. However, the `normal.splits` only guarantees the splitting of minpolys of elements in E; since `adjoin F S` embeds in `E` (via `φ`), the minpolys of elements in `adjoin F S` also splits because they are equal to the minpolys of the images of the elements in `E`. This is what `rw [minpoly.algHom_eq _ φ.injective]` does, and then we need another rewrite to go between `adjoin F S` and `K`.
The whole proof is a bit nontrivial and uses `Normal.of_isSplittingField` essentially.